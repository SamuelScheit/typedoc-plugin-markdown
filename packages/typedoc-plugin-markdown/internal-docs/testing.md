---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
