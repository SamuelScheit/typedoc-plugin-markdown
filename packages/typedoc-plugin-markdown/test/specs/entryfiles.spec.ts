import { expectFileToEqual } from '@devtools/testing';

describe(`EntryFiles`, () => {
  test(`should get entry module`, () => {
    expectFileToEqual('entryfiles', 'members', ['README.md', 'index.md']);
  });

  test(`should get module member for members`, () => {
    expectFileToEqual(
      'entryfiles',
      'members',
      'entry-module/classes/ModuleClass.md',
    );
  });

  test(`should get readme file`, () => {
    expectFileToEqual('entryfiles', 'members', 'readme_.md', 1);
  });
});
