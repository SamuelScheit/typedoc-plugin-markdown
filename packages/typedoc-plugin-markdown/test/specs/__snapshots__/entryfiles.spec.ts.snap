// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryFiles should get entry module: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** • [Readme](readme_.md) \\| API

***

# entry-module

## Index

### Classes

- [ModuleClass](entry-module/classes/ModuleClass.md)
"
`;

exports[`EntryFiles should get entry module: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs** • API

***

# entry-module

## Index

### Classes

- [ModuleClass](entry-module/classes/ModuleClass.md)
"
`;

exports[`EntryFiles should get module member for members: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** • [Readme](../../readme_.md) \\| [API](../../README.md)

***

[typedoc-stubs](../../README.md) / ModuleClass

# Class: ModuleClass

## Constructors

### new ModuleClass()

• **new ModuleClass**(): [\`ModuleClass\`](ModuleClass.md)

#### Returns

[\`ModuleClass\`](ModuleClass.md)
"
`;

exports[`EntryFiles should get module member for members: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs** • API

***

[typedoc-stubs](../../index.md) / ModuleClass

# Class: ModuleClass

## Constructors

### new ModuleClass()

• **new ModuleClass**(): [\`ModuleClass\`](ModuleClass.md)

#### Returns

[\`ModuleClass\`](ModuleClass.md)
"
`;

exports[`EntryFiles should get readme file: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** • Readme \\| [API](README.md)

***

# MY README

> Some block quote

Some text.

## Readme heading

Some text.
"
`;
